src/wp-includes/fonts/class-wp-font-collection.php

```diff
@@ -0,0 +1,260 @@
+	 */
+	private function load_from_url( $url ) {
+		// Limit key to 167 characters to avoid failure in the case of a long URL.
+		$transient_key = substr( 'wp_font_collection_url_' . $url, 0, 167 );
+		$data          = get_site_transient( $transient_key );
+
+		if ( false === $data ) {
+			$response = wp_safe_remote_get( $url );
+			if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+				// translators: %s: Font collection URL.
+				return new WP_Error( 'font_collection_request_error', sprintf( __( 'Error fetching the font collection data from "%s".' ), $url ) );
+			}
+
+			$data = json_decode( wp_remote_retrieve_body( $response ), true );
+			if ( empty( $data ) ) {
+				return new WP_Error( 'font_collection_decode_error', __( 'Error decoding the font collection data from the HTTP response JSON.' ) );
+			}
+
+			// Make sure the data is valid before storing it in a transient.
+			$data = $this->sanitize_and_validate_data( $data );
+			if ( is_wp_error( $data ) ) {
+				return $data;
+			}
+
+			set_site_transient( $transient_key, $data, DAY_IN_SECONDS );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Sanitizes and validates the font collection data.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $data Font collection data to sanitize and validate.
+	 * @return array|WP_Error Sanitized data if valid, otherwise a WP_Error instance.
+	 */
+	private function sanitize_and_validate_data( $data ) {
+		$schema = self::get_sanitization_schema();
+		$data   = WP_Font_Utils::sanitize_from_schema( $data, $schema );
+
+		$required_properties = array( 'name', 'font_families' );
+		foreach ( $required_properties as $property ) {
+			if ( empty( $data[ $property ] ) ) {
+				$message = sprintf(
+					// translators: 1: Font collection slug, 2: Missing property name, e.g. "font_families".
+					__( 'Font collection "%1$s" has missing or empty property: "%2$s".' ),
+					$this->slug,
+					$property
+				);
+				_doing_it_wrong( __METHOD__, $message, '6.5.0' );
+				return new WP_Error( 'font_collection_missing_property', $message );
+			}
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Retrieves the font collection sanitization schema.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array Font collection sanitization schema.
+	 */
+	private static function get_sanitization_schema() {
+		return array(
+			'name'          => 'sanitize_text_field',
+			'description'   => 'sanitize_text_field',
+			'font_families' => array(
+				array(
+					'font_family_settings' => array(
+						'name'       => 'sanitize_text_field',
+						'slug'       => 'sanitize_title',
+						'fontFamily' => 'sanitize_text_field',
+						'preview'    => 'sanitize_url',
+						'fontFace'   => array(
+							array(
+								'fontFamily'            => 'sanitize_text_field',
+								'fontStyle'             => 'sanitize_text_field',
+								'fontWeight'            => 'sanitize_text_field',
+								'src'                   => function ( $value ) {
```

Comment:
Teeny tiny micro-optimization

```suggestion
								'src'                   => static function ( $value ) {
```

Reply:
Addressing in Gutenberg here: https://github.com/WordPress/gutenberg/pull/58691/commits/4d8472acac32ef24071fdd5edbe295f964d363c6

Reply:
Changed in https://github.com/WordPress/wordpress-develop/pull/6027/commits/095fd857413b24fd674284cd3229e4100cf0d994 ✅